src/frontend/components/JoinDiscussionAskButton/index.tsx

```diff
@@ -28,18 +29,46 @@ const messages = defineMessages({
 });
 
 export const JoinDiscussionAskButton = () => {
-  const { asked, accepted, rejected, setAsked } = useParticipantWorkflow(
-    (state) => ({
-      asked: state.asked,
-      accepted: state.accepted,
-      rejected: state.rejected,
-      setAsked: state.setAsked,
-    }),
-  );
+  const [username, setUsername] = useState('');
+  const [showUsernameForm, setShowUsernameForm] = useState(false);
+  const {
+    asked,
+    accepted,
+    rejected,
+    usernameAlreadyExisting,
+    reset,
+    setAsked,
+  } = useParticipantWorkflow((state) => ({
+    asked: state.asked,
+    accepted: state.accepted,
+    rejected: state.rejected,
+    usernameAlreadyExisting: state.usernameAlreadyExisting,
+    setAsked: state.setAsked,
+    reset: state.reset,
+  }));
+
+  useEffect(() => {
+    if (usernameAlreadyExisting) {
+      setShowUsernameForm(true);
+    }
+  }, [usernameAlreadyExisting]);
+
+  const closeLayer = () => {
+    setUsername('');
+    setShowUsernameForm(false);
+    reset();
+  };
 
-  const onClick = () => {
-    converse.askParticipantToJoin();
-    setAsked();
+  const onClick = async () => {
+    try {
+      await converse.askParticipantToJoin(username);
+      setAsked();
+      if (showUsernameForm) {
+        setShowUsernameForm(false);
+      }
+    } catch (error) {
+      setShowUsernameForm(true);
```

Comment:
Hmm, `showUsernameForm` will be ever equal to `true` at this time

Reply:
Can you explain me why ? the `askParticipantToJoin` function throws an error if the user is not connected and at this time the form is shown to choose a username to connect him.

Reply:
Because `onClick` function is called when `showUsernameForm=true` and if `askParticipantToJoin` throws an error, `setShowUsernameForm(false)` is not reached.

Reply:
`onClick` function can be called also when `showUsernameForm=false` line 151 on the `ask` button.

Reply:
Oh I missed that ! 🤦‍♂️

Reply:
Should I create 2 distinct function ?

Reply:
Nope. As in fact, as state did not change, Reat will not rerender the component.